### ISSUE TITLE
Has anyone used Primus with websockets behind aws Elastic Load Balancer?

### ISSUE THREAD

**Issue by manikandants** [primus/primus#438]
**Version:** 4.0.5

**Environment:** AWS Elasticbeanstalk
  - **Operating system:** Amazon linux 64bit
  - **Browser:**
  - **Node.js:** 0.12.6

**Expected result:**

**Actual result:**

**Steps to reproduce:**
I have a node.js application server and I recently added realtime messaging through Primus websockets transformer. It works well in single instance. I deployed the messaging to production environment (AWS Elastic beanstalk) and added websocket port to security group. But the client requests doesn't reach the instance. Has anyone used primus with websockets behind AWS ELB? Please let me know what needs to be done to enable the communication

**Comment by lpinca** [comment]
AFAIK, AWS ELB does not support WebSocket.

**Comment by manikandants** [comment]
Thanks @lpinca That's a bad news for me :( How is the support for any other transformers like Socket.io/Sockjs/Engine.io?

**Comment by lpinca** [comment]
It might work but you'll be limited to the polling transports. Take a look at this old issue https://github.com/primus/primus/issues/87

**Comment by manikandants** [comment]
Thanks for the info. I'm trying out the proxywrap module. I'll explore and update on the issue.

**Comment by ddennis** [comment]
So i might be misunderstanding the issue, sorry if i am way off.

But I have been running the WS module on Elastic Beanstalk with Primus in production for over a year, without any problems. Using the node cluster module to scale out. 

Have you tried setting the proxy server to "none":
On Elastic Beanstalk goto Configuration > Software Configuration > Container Options: set Proxy server: none.

By the way i have had problems installing the ws module on newer version of node on Elastic Beanstalk, so i would recommend staying on node 0.12.6 and ELB node v2.0.8

**Comment by manikandants** [comment]
@ddennis My proxy server is set to "none" in container options.
I am still on ELB v2.0.0 running nodejs 0.12.6.
Have you enabled proxy-protocol as mentioned in [aws documentation](http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/enable-proxy-protocol.html#proxy-protocol-prerequisites)?
What is the listener configuration and security group configuration that is working for you?
I have application server running on port 80 and websockets running on port 9001. I have opened both ports in security group, added TCP listener for port 9001, enabled proxy protocol as mentioned in the link above and included proxywrap as well. But still client is not able to connect to websocket. I probably am missing some silly thing. but couldn't figure out the same.

**Comment by manikandants** [comment]
You can find my configuration with screenshots in [stackoverflow](http://stackoverflow.com/questions/35713976/has-anyone-used-primus-with-websockets-behind-aws-elastic-load-balancer)

**Comment by ddennis** [comment]
@manikandants I have not enabled the proxy-protocol, or included proxywrap.

Configuration > Software Configuration > Container Options
![config](https://cloud.githubusercontent.com/assets/300269/13425038/abc7e464-dfa4-11e5-9d8a-b3cbfea66525.png)


TCP Listener is added in the Elastic Load Balancer configuration. 
![lb](https://cloud.githubusercontent.com/assets/300269/13424991/585d2a96-dfa4-11e5-9ebb-c7a90ca1faef.png)

security group of the instance 
![inbound](https://cloud.githubusercontent.com/assets/300269/13425013/748b61ba-dfa4-11e5-9ca0-d25cdcb0882a.png)

**Comment by manikandants** [comment]
Thanks. I'll disable proxy and try it.

**Comment by lpinca** [comment]
Thank you for the help @ddennis. @manikandants any luck so far?

**Comment by manikandants** [comment]
Nope. The requests are reaching the loadbalancer, but not the ec2 instances behind it. We have temporarily created an ec2 instance just to handle messaging, along with the load balanced application server until we figure out the issue/our userbase get big enough to overload the ec2 instance.

**Comment by codepreneur** [comment]
@manikandants I managed to get websockets (https://github.com/websockets/ws) working on both port 80 and 8080 on ELB with the configurations below.

Security group:

![image](https://cloud.githubusercontent.com/assets/3421858/14969922/ff6659f4-10bc-11e6-903c-168a530efebd.png)


Load balancer listeners:

![image](https://cloud.githubusercontent.com/assets/3421858/14969937/160c7544-10bd-11e6-8357-a11f5481d9fd.png)


Container options:

![image](https://cloud.githubusercontent.com/assets/3421858/14969950/38776f1c-10bd-11e6-961b-aabbf0ab1d66.png)

**Comment by alka-axonic** [comment]
Hello, i have same Problem. Wir have nodejs App with Primus. If I try to make tcp connection over Port 80 and 443 works sockets but the page dies not work. If I try to make the connection over https work the page, but sockets does no work. I try to activate Proxy protocol, does not help. Has anyone ideas?

**Comment by djamalouch** [comment]
AWS not seems to support Websockets: https://aws.amazon.com/blogs/aws/new-aws-application-load-balancer/
